packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.ts

```diff
@@ -202,111 +164,90 @@ async function askApiNames(context, defaults) {
-  ]);
+  const resourceName = await prompter.input<'one', string>(
+    'Provide a friendly name for your resource to be used as a label for this category in the project:',
+    { initial: defaultResourceName, validate: apiNameValidator },
+  );
 
-  answer.apiName = answer.resourceName;
-
-  return answer;
+  return resourceName;
 }
 
-async function askPrivacy(context, answers, currentPath) {
+async function askPermissions(
+  context: $TSContext,
+  answers: $TSObject,
+  currentPath?: ApigwPath,
+): Promise<{ setting?: PermissionSetting; auth?: CrudOperation[]; open?: boolean; userPoolGroups?: $TSObject; guest?: CrudOperation[] }> {
   while (true) {
-    const apiAccess = await inquirer.prompt({
-      name: 'restrict',
-      type: 'confirm',
-      default: !(currentPath && currentPath.open),
-      message: 'Restrict API access',
-    });
+    const apiAccess = await prompter.yesOrNo('Restrict API access', currentPath?.permissions?.setting !== PermissionSetting.OPEN);
 
-    if (!apiAccess.restrict) {
-      return { open: true };
+    if (!apiAccess) {
+      return { setting: PermissionSetting.OPEN };
     }
 
-    const userPoolGroupList = await context.amplify.getUserPoolGroupList(context);
+    const userPoolGroupList = context.amplify.getUserPoolGroupList();
 
     let permissionSelected = 'Auth/Guest Users';
-    const privacy: any = {};
+    const permissions: $TSObject = {};
 
     if (userPoolGroupList.length > 0) {
       do {
         if (permissionSelected === 'Learn more') {
-          context.print.info('');
-          context.print.info(
-            'You can restrict access using CRUD policies for Authenticated Users, Guest Users, or on individual Group that users belong to in a User Pool. If a user logs into your application and is not a member of any group they will use policy set for “Authenticated Users”, however if they belong to a group they will only get the policy associated with that specific group.',
+          printer.blankLine();
+          printer.info(
+            'You can restrict access using CRUD policies for Authenticated Users, Guest Users, or on individual Group that users belong to' +
+              ' in a User Pool. If a user logs into your application and is not a member of any group they will use policy set for ' +
+              '“Authenticated Users”, however if they belong to a group they will only get the policy associated with that specific group.',
           );
-          context.print.info('');
+          printer.blankLine();
         }
-        const permissionSelection = await inquirer.prompt({
-          name: 'selection',
-          type: 'list',
-          message: 'Restrict access by?',
-          choices: ['Auth/Guest Users', 'Individual Groups', 'Both', 'Learn more'],
-          default: 'Auth/Guest Users',
-        });
-
-        permissionSelected = permissionSelection.selection;
+        const permissionSelection = await prompter.pick<'one', string>('Restrict access by?', [
```

Comment:
```suggestion
        const permissionSelection = await prompter.pick<'one', string>('Restrict access by:', [
```

Reply:
This change should be fine to merge, the tests do not wait on the `?`